src/stack/tcp.rs

```diff
@@ -1,5 +1,22 @@
 use crate::SocketAddr;
 
+/// Represents specific errors encountered during TCP operations.
+#[non_exhaustive]
+#[derive(Copy, Clone, PartialEq, Debug)]
+pub enum TcpErrorKind {
+	/// The peer has closed one or both directions and the connection is broken.
```

Comment:
Could this be more precise, eg. like this?

```suggestion
	/// The socket has been closed in the direction in which the failing operation was attempted.
```

This would
* make no implications on who closed the connection (we currently don't have API to create a half-closed connection from our end, but maybe something else caused the OS to close our stream, not sure if realistic), and
* tell which direction was affected (because one-or-both kind of implies that a write may fail if the inbound half was closed)

Reply:
I like this change for sure! Nice clarification :)